File: mergesort.py
import time
print("Enter the elements of the array")
array=list(map(int,input().split()))
size=len(array)
def merge(low,mid,high):
    temp=[0 for x in range(high+1)]
    i=low
    j=mid+1
    k=low
    while i<=mid and j<=high:
        if array[i]>array[j]:
            temp[k]=array[j]
            k+=1
            j+=1
        else:
            temp[k]=array[i]
            k+=1
            i+=1
    while i<=mid:
        temp[k]=array[i]
        k+=1
        i+=1
    while j<=high:
        temp[k]=array[j]
        k+=1
        j+=1
    for index in range(low,high+1):
        array[index]=temp[index]
def mergesort(low,high):
    if low<high:
        mid=(low+high)//2
        mergesort(low,mid)
        mergesort(mid+1,high)
        merge(low,mid,high)
mergesort(0,size-1)
print(array)

File: merge_sort.cpp
#include<iostream>
#include<time.h>
using namespace std;
int *array;
void merge(int low,int mid,int high)
{
    int i=low,j=mid+1,k=low;
    int temp[high+1];
    while(i<=mid && j<=high)
    {
        if(array[i]>array[j])
            temp[k++]=array[j++];
        else
            temp[k++]=array[i++];
    }
    for(;i<=mid;i++,k++)
        temp[k]=array[i];
    for(;j<=high;j++,k++)
        temp[k]=array[j];
    for(i=low;i<=high;i++)
        array[i]=temp[i];
}
void merge_sort(int low,int high)
{
    if(low<high)
    {
        int mid = (low+high)/2;
        merge_sort(low,mid);
        merge_sort(mid+1,high);
        merge(low,mid,high);    
    }
}
int main()
{
    clock_t time;
    int size;
    cout<<"Enter the size of the array \n>>";
    cin>>size;
    array=new int[size];
    cout<<"Enter the elements of the array"<<endl;
    for(int index=0;index<size;index++)
        cin>>array[index];
    time=clock();
    merge_sort(0,size-1);
    time=(clock()-time)/(double)CLOCKS_PER_SEC;
    for(int index=0;index<size;index++)
        cout<<array[index]<<" ";
    cout<<endl;
    cout<<time<<endl;
    return 0;
}
